File: src/Modules/InstantMessenger.Groups/Domain/Messages/Entities/Message.cs
using System;
using InstantMessenger.Groups.Domain.Entities;
using InstantMessenger.Groups.Domain.Events;
using InstantMessenger.Groups.Domain.Messages.ValueObjects;
using InstantMessenger.Groups.Domain.ValueObjects;
using InstantMessenger.Shared.BuildingBlocks;

namespace InstantMessenger.Groups.Domain.Messages.Entities
{
    public sealed class Message : Entity<MessageId>
    {
        public GroupId GroupId { get; }
        public ChannelId ChannelId { get; }
        public UserId From { get; }
        public MessageContent Content { get; }

        public DateTimeOffset CreatedAt { get; }

        public Message(MessageId id, GroupId groupId, ChannelId channelId, UserId from, MessageContent content, DateTimeOffset createdAt):base(id)
        {
            ChannelId = channelId;
            From = from;
            Content = content;
            CreatedAt = createdAt;
            GroupId = groupId;
            Apply(new MessageCreatedEvent(Id,GroupId,ChannelId, From, Content,CreatedAt));
        }
    }
}

File: src/InstantMessenger.Web/ClientApp/src/app/home/groups/store/types/message.ts
export interface ChannelMessageDto {
  channelId: string;
  groupId: string;
  message: MessageDto[];
}
export interface MessageDto {
  channelId: string;
  groupId: string;
  messageId: string;
  senderId: string;
  senderName: string;
  senderAvatar: string;
  content: string;
  createdAt: string;
}
